opennms-services/src/main/java/org/opennms/netmgt/poller/ServiceMonitorCamelImpl.java

```diff
@@ -30,5 +29,9 @@ public void close() {
 	public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
```

Comment:
I think you will need to get rid of this method and change the ServiceMonitor interface so that the only poll method is `poll(MonitoredServiceTask)`.

You need to add a `MonitoredServiceTask.getLocation()` call so that the tasks can be routed to the proper JMS queue, for example `activemq:Location-${body.location}.Poller.HttpMonitor`.

You need to add a `MonitoredServiceTask.calculateTaskTimeout()` method that is similar to `org.opennms.netmgt.discovery.messages.DiscoveryJob.calculateTaskTimeout()` as well so that we can use that value to set the JMS message timeout.